### ISSUE TITLE
Libmaxminddb doesn't support file paths outside the current code page on Windows.

### ISSUE THREAD

**Issue by geraldcombs** [maxmind/libmaxminddb#189]
If I create an account for [Bruce Lee](https://en.wikipedia.org/wiki/Bruce_Lee) on a Windows 10 system here such that I have a directory named `C:\Users\李小龍` and place GeoLite2-City.mmdb in `C:\Users\李小龍\GeoIP`, mmdblookup returns the following error:

```
> .\mmdblookup --file C:\Users\李小龍\GeoIP\GeoLite2-City.mmdb --ip 8.8.8.8

  Can't open C:\Users\???\GeoIP\GeoLite2-City.mmdb - Error opening the specified MaxMind DB file
```

This appears to be due to libmaxminddb using the narrow (ANSI) version of the Win32 API, specifically CreateFileA:

https://github.com/maxmind/libmaxminddb/blob/master/src/maxminddb.c#L331

Using the ANSI API goes against Microsoft's recommendations. As https://docs.microsoft.com/en-us/windows/desktop/intl/conventions-for-function-prototypes says,

"New Windows applications should use Unicode to avoid the inconsistencies of varied code pages and for ease of localization. They should be written with generic functions, and should define UNICODE to compile the functions into Unicode functions. In the few places where an application must work with 8-bit character data, it can make explicit use of the functions for Windows code pages."

There are a few ways to fix the problem:

1) Ignore it. Sorry, Bruce.

2) Specify that MMDB_open takes a narrow, UTF-8 encoded file path on Windows and update the libmaxminddb and mmdbresolve code accordingly. This would match the behavior of other cross-platform libraries such as GLib and is the solution I'd personally recommend.

3) Specify that MMDB_open takes a wide, UTF-16LE encoded file path on Windows and update the libmaxminddb and mmdbresolve code accordingly.

I'd be glad to help implement 2).

**Comment by metafex** [comment]
Wow, that was five years ago, I don't remember why it used the A variant... So yes, I'd also suggest solution 2, it's the most sensible option.

**Comment by geraldcombs** [comment]
Proposed fix in #191.

**Comment by oschwald** [comment]
Fixed in #191.
